client/asset/btc/btc.go

```diff
@@ -987,49 +1011,196 @@ func (btc *ExchangeWallet) PreSwap(req *asset.PreSwapForm) (*asset.PreSwap, erro
+				},
+				End: asset.XYRangePoint{
+					Label: bumpLabel,
+					X:     maxBump,
+					Y:     float64(req.FeeSuggestion) * maxBump,
+				},
+				XUnit: "X",
+				YUnit: btc.walletInfo.UnitInfo.AtomicUnit + "/" + btc.sizeUnit(),
+			},
+		})
+	}
+
 	return &asset.PreSwap{
 		Estimate: est,
+		Options:  opts,
 	}, nil
 }
 
+// splitOption constructs an *asset.OrderOption with customized text based on the
+// difference in fees between the configured and test split condition.
+func (btc *ExchangeWallet) splitOption(req *asset.PreSwapForm, utxos []*compositeUTXO, bump float64) *asset.OrderOption {
+	noSplitEst, _, noSplitLocked, err := btc.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, false, bump)
+	if err != nil {
+		btc.log.Errorf("estimateSwap (no split) error: %v", err)
+		return nil
+	}
+	splitEst, splitUsed, splitLocked, err := btc.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, true, bump)
+	if err != nil {
+		btc.log.Errorf("estimateSwap (with split) error: %v", err)
+		return nil
+	}
+	if !splitUsed {
+		// unable to do the split. no option.
+		btc.log.Debugf("split option unavailable")
+		return nil
+	}
+	symbol := strings.ToUpper(btc.symbol)
+
+	xtraFees := splitEst.RealisticWorstCase - noSplitEst.RealisticWorstCase
+	pctChange := (float64(splitEst.RealisticWorstCase)/float64(noSplitEst.RealisticWorstCase) - 1) * 100
+	overlock := noSplitLocked - splitLocked
+
+	var reason string
+	if pctChange > 1 {
+		reason = fmt.Sprintf("+%d%% fees, -%s %s overlock", int(math.Round(pctChange)), prettyBTC(overlock), symbol)
+	} else {
+		reason = fmt.Sprintf("+%.1f%% fees, -%s %s overlock", pctChange, prettyBTC(overlock), symbol)
+	}
+
+	desc := fmt.Sprintf("Using a split transaction to prevent temporary overlock of %s %s, but for additional fees of %s %s",
+		prettyBTC(overlock), symbol, prettyBTC(xtraFees), symbol)
+
+	return &asset.OrderOption{
+		ConfigOption: asset.ConfigOption{
+			Key:          splitKey,
+			DisplayName:  "Pre-size Funds",
+			Description:  desc,
+			DefaultValue: btc.useSplitTx,
+			IsBoolean:    true,
+		},
+		Boolean: &asset.BooleanConfig{
+			Reason: reason,
+		},
+	}
+}
+
 // estimateSwap prepares an *asset.SwapEstimate.
 func (btc *ExchangeWallet) estimateSwap(lots, lotSize, feeSuggestion uint64, utxos []*compositeUTXO,
-	nfo *dex.Asset, trySplit bool) (*asset.SwapEstimate, bool /*split used*/, error) {
+	nfo *dex.Asset, trySplit bool, feeBump float64) (*asset.SwapEstimate, bool /*split used*/, uint64 /*amt locked*/, error) {
 
 	var avail uint64
 	for _, utxo := range utxos {
 		avail += utxo.amount
 	}
 
+	// If there is a fee bump, the networkFeeRate can be higher than the
+	// MaxFeeRate
+	bumpedMaxRate := nfo.MaxFeeRate
+	bumpedNetRate := feeSuggestion
+	if feeBump > 1.01 {
+		bumpedMaxRate = uint64(math.Round(float64(bumpedMaxRate) * feeBump))
+		bumpedNetRate = uint64(math.Round(float64(bumpedNetRate) * feeBump))
+	}
```

Comment:
Sort of setting an informal minimum bump of 1% here, but there's not real reason it has to be that way.

Reply:
Greater than 1% looks like.  But I don't see a downside to blindly multiplying by `feeBump` as long as it is >1.  

More fundamentally, the units and meaning of `FeeBump` all the way back to `swapOptions` and `redeemOptions` is not perfectly clear.  It could conceivable be a percent added e.g. 0.01 instead of 1.01, or it could even be an integer that is an absolute sat/vB, or an integer that represents steps of say 0.1%.